### PR TITLE
Fixing race issue

### PR DESCRIPTION
**Pull request by udai-arneja**
Removed the deletion to after iteration.

# Summary

Fixing RACE issue - moved the deletion to after iteration.

## Additional Information

Solving issue #93 

## Test Plan

Ran over prolonged period of days. All seems to be clear and making sense

### CODE REVIEW

**Comment by udai-arneja** [comment]
@jzzheng22 will wait on ur review before merging X

File: pkg/simulation/simIteration.go
Comment (jzzheng22) [inline pkg/simulation/simIteration.go:57]: No use in having a mutex here

File: pkg/infra/baseagent.go
Comment (jzzheng22) [inline pkg/infra/baseagent.go:88]: Don't think this will work since a.hp is never set to 0 I think

**Review by jzzheng22 — CHANGES_REQUESTED** [review]

File: pkg/infra/baseagent.go
Comment (udai-arneja) [reply pkg/infra/baseagent.go]: V true hahaa

**Comment by udai-arneja** [comment]
Closing due to better overall solution.

**Pull request closed without merging**
